### PR TITLE
[Akeyless] Adding GJSON pkg support to extracting secret key value from JSON

### PR DESCRIPTION
**Pull request by charan986**
## Problem Statement
Current ESO Akeyless is unable to fetch a secret key value from a JSON secret
What is the problem you're trying to solve?
fetching a secret key value from a JSON secret
## Related Issue

Fixes #...

## Proposed Changes

How do you like to solve the issue and why?

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [ ] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [ ] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by moolen** [comment]
@renanaAkeyless could you please take a look at this PR? :pray:

**Comment by renanaAkeyless** [comment]
> @renanaAkeyless could you please take a look at this PR? 🙏

sure

File: pkg/provider/akeyless/akeyless.go [hunk 1]
```diff
@@ -256,7 +257,27 @@ func (a *Akeyless) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDa
...
+	if ref.Property == "" {
+		if value != "" {
+			return []byte(value), nil
+		}
+		return nil, fmt.Errorf("invalid value received, found no value string : %s", ref.Key)
+	}
+	// We need to search if a given key with a . exists before using gjson operations.
+	idx := strings.Index(ref.Property, ".")
+	if idx > -1 {
+		refProperty := strings.ReplaceAll(ref.Property, ".", "\\.")
```
Comment (renanaAkeyless) [inline pkg/provider/akeyless/akeyless.go:270]: why do we need the "replace" in this case? according to the documentation,  it should work out of the box (https://github.com/tidwall/gjson#get-a-value).

@charan986  did you manually test it?

Comment (charan986) [reply pkg/provider/akeyless/akeyless.go:270]: @renanaAkeyless - I manually tested all the JSON use cases that needs to be fetched using these changes

Below is the JSON key "b.cd" test done for the above 
{
  "abc": "mega-value \n",
  "b.cd": "Wahoo!"
}

Also for reference, We followed how aws was trying to fetch the JSON key values

https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/aws/secretsmanager/secretsmanager.go#L411-L417

**Review by renanaAkeyless — APPROVED** [review]

**Review by moolen — APPROVED** [review]

**Pull request merged**